test/client_test.js

```diff
@@ -57,26 +57,95 @@ describe('Client', () => {
         .get('/Patient/eb3271e1-ae1b-4644-9332-41e32c829486')
         .reply(200, () => readStreamFor('patient.json'));
 
-      const response = await this.fhirClient.read('Patient', 'eb3271e1-ae1b-4644-9332-41e32c829486');
+      const response = await this.fhirClient.read({ resourceType: 'Patient', identifier: 'eb3271e1-ae1b-4644-9332-41e32c829486' });
 
       expect(response.resourceType).to.equal('Patient');
       expect(response.id).to.equal('eb3271e1-ae1b-4644-9332-41e32c829486');
     });
 
-    it('responds to #read, throwing any errors that occur', async function () {
+    it('responds to #read, throwing any errors for a missing resource', async function () {
       nock(baseUrl)
         .matchHeader('accept', 'application/json+fhir')
         .get('/Patient/abcdef')
         .reply(404, () => readStreamFor('patient-not-found.json'));
 
       let response;
       try {
-        response = await this.fhirClient.read('Patient', 'abcdef');
+        response = await this.fhirClient.read({ resourceType: 'Patient', identifier: 'abcdef' });
       } catch (error) {
         expect(error.response.status).to.equal(404);
         expect(error.response.data.resourceType).to.deep.equal('OperationOutcome');
       }
       expect(response).to.be.undefined; // eslint-disable-line no-unused-expressions
     });
+
+    it('responds to #vread, returning a matching resource', async function () {
+      nock(baseUrl)
+        .matchHeader('accept', 'application/json+fhir')
+        .get('/Patient/eb3271e1-ae1b-4644-9332-41e32c829486/_history/1')
+        .reply(200, () => readStreamFor('patient.json'));
+
+      const response = await this.fhirClient.vread({ resourceType: 'Patient', identifier: 'eb3271e1-ae1b-4644-9332-41e32c829486', version: '1' });
+
+      expect(response.resourceType).to.equal('Patient');
+      expect(response.id).to.equal('eb3271e1-ae1b-4644-9332-41e32c829486');
+    });
+
+    it('responds to #vread, throwing any errors for an absent resource', async function () {
+      nock(baseUrl)
+        .matchHeader('accept', 'application/json+fhir')
+        .get('/Patient/abcdef/_history/1')
+        .reply(404, () => readStreamFor('patient-not-found.json'));
+
+      let response;
+      try {
+        response = await this.fhirClient.vread({ resourceType: 'Patient', identifier: 'abcdef', version: '1' });
+      } catch (error) {
+        expect(error.response.status).to.equal(404);
+        expect(error.response.data.resourceType).to.deep.equal('OperationOutcome');
+      }
+      expect(response).to.be.undefined; // eslint-disable-line no-unused-expressions
+    });
+
+    it('responds to #vread, throwing any errors for an absent version of an existing resource', async function () {
```

Comment:
I don't think this test really adds anything that isn't in the test above. From the client's perspective, it received an http error code in both cases. It doesn't care what specific error code it was, or what in particular caused the error.

I know it's normal not to test private methods, but I wonder if it would be best to just test the error throwing on `httpGet` so that we don't have to add these tests for every FHIR action.